types/mdx/types.d.ts

```diff
@@ -1,12 +1,55 @@
 // Internal helper types
 
-// tslint:disable-next-line: strict-export-declare-modifiers
-type FunctionComponent<Props> = (props: Props) => JSX.Element | null;
-// tslint:disable-next-line: strict-export-declare-modifiers
-type ClassComponent<Props> = new (props: Props) => JSX.ElementClass;
-// tslint:disable-next-line: strict-export-declare-modifiers
-type Component<Props> = FunctionComponent<Props> | ClassComponent<Props> | keyof JSX.IntrinsicElements;
-// tslint:disable-next-line: strict-export-declare-modifiers
+/**
+ * This is the global JSX.ElementType if it’s defined, otherwise never.
+ */
+// @ts-ignore
+type ElementType = any extends JSX.ElementType ? never : JSX.ElementType;
```

Comment:
@jakebailey Pinging a random DefinitelyTyped maintainer.

This construct is needed to make the types work for frameworks that may or may not define the `JSX.ElementType` type. It works as-is, but CI fails on the `@ts-ignore` comment. It’s not just some lint rule I can disable. Do you have any suggestions?

Reply:
I'm not sure how I feel about this... There's no `@ts-ignore` in the entire DT repo and I feel like it should stay that way.

Reply:
Probably @sandersn or @andrewbranch have a better idea of what can be done here.

Reply:
I had this sitting in my inbox but forgot until the scary bot message; tomorrow I'll go see about switching this to the eslint rule instead.

Reply:
Alright, this should all be done and merged. You should be able to merge `master` and re-npm-install.

Reply:
Note that the rule will require you to put a short explanation of why the ts-ignore is needed, which is good practice in my books.